portal-bridge/src/types/era1.rs

```diff
@@ -66,21 +67,25 @@ impl Era1 {
     pub fn deserialize(buf: &[u8]) -> anyhow::Result<Self> {
         let file = E2StoreFile::deserialize(buf)?;
         ensure!(
-            file.entries.len() == ERA1_ENTRY_COUNT,
+            // era1 file #0-1895 || era1 file #1896
+            file.entries.len() == ERA1_ENTRY_COUNT || file.entries.len() == 21451,
```

Comment:
If I understand correctly, I'm not sure this is the place for that kind of verification.
The hash that this would be verified against does exists in the file path we provide, however there's no guarantee that this filepath is always the pointer provided for the file contents
- you can deserialize from a byte vector, so you wouldn't have a file path
- you can load an era1 file from a different filepath
So if there was to be any checking of the hash, it should be done by code/tooling a level above this module.
The same applies to index number & first/last block number. This code doesn't know what index/block number it's expected to find, so that validation should also occur a level above. I guess we could do some validation that the `first_block_number = last_block_number - 8192`, but I'm not sure if that's necessary? It's the kind of thing that would be covered if we're checking the root matches the expected hash a level above.